big_tests/tests/mam_SUITE.erl

```diff
@@ -2194,6 +2195,52 @@ muc_archive_request(Config) ->
         end,
     escalus:story(Config, [{alice, 1}, {bob, 1}], F).
 
+muc_archive_request_after_mam_id_reset(Config) ->
+    P = ?config(props, Config),
+    F = fun(Alice, Bob) ->
+        StartTS = erlang:system_time(microsecond),
+        Room = ?config(room, Config),
+        RoomAddr = room_address(Room),
+        Text = <<"Hi, Bob!">>,
+        escalus:send(Alice, stanza_muc_enter_room(Room, nick(Alice))),
+        escalus:send(Bob, stanza_muc_enter_room(Room, nick(Bob))),
+
+        %% Bob received presences.
+        escalus:wait_for_stanzas(Bob, 2),
+
+        %% Bob received the room's subject.
+        escalus:wait_for_stanzas(Bob, 1),
+
+        %% Alice sends another message to Bob.
+        %% The message is not archived by the room.
+        escalus:send(Alice, escalus_stanza:chat_to(Bob, <<"OH, HAI!">>)),
+        escalus:assert(is_message, escalus:wait_for_stanza(Bob)),
+
+        %% Alice sends to the chat room.
+        escalus:send(Alice, escalus_stanza:groupchat_to(RoomAddr, Text)),
+        0 = rpc(mim(), mongoose_mam_id, reset, []),
```

Comment:
So the whole NIF was added just to call it here? How does this correspond to a real use case? This looks so artificial to me, that I don't think we need this test case...

Reply:
non artificial way would be:
- wait a couple of minutes without doing anything.
- send a message.
- message would have timestamp 2 minutes ago.

Which obviously we do not want to do on CI.
Mock? yes, I can rewrite test with a mock.